pkg/broker/user_info.go

```diff
@@ -0,0 +1,78 @@
+package broker
+
+import (
+	"encoding/json"
+	"fmt"
+
+	osb "github.com/pmorie/go-open-service-broker-client/v2"
+)
+
+// parseKubernetesIdentity - creates a kubernetes identity from the
+// orginating identity
+func parseKubernetesIdentity(o osb.OriginatingIdentity) (*UserInfo, error) {
+	u := UserInfo{}
+	err := json.Unmarshal([]byte(o.Value), &u)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal json for value while parsing Kubernetes identity")
+	}
+	return &u, nil
+}
+
+// parseCloudFoundryIdentity - creates a cloud foundry identity from the
+// orginating identity
+func parseCloudFoundryIdentity(o osb.OriginatingIdentity) (map[string]interface{}, error) {
+	m := map[string]interface{}{}
+	err := json.Unmarshal([]byte(o.Value), &m)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal json for value while parsing cloud foundry identity")
+	}
+	// Validate that user_id MUST be in the json object.
+	if _, ok := m["user_id"]; !ok {
+		return nil, fmt.Errorf("user_id key was not found in cloud foundry object")
+	}
+	return m, nil
+}
+
+// ParseIdentity - retrieve the identity union type
+func ParseIdentity(o osb.OriginatingIdentity) (Identity, error) {
+	identity := Identity{Platform: o.Platform}
+	switch o.Platform {
+	case osb.PlatformKubernetes:
+		k, err := parseKubernetesIdentity(o)
+		if err != nil {
+			return identity, err
+		}
+		identity.Kubernetes = k
+	case osb.PlatformCloudFoundry:
+		c, err := parseCloudFoundryIdentity(o)
+		if err != nil {
+			return identity, err
+		}
+		identity.CloudFoundry = c
+	default:
+		m := map[string]interface{}{}
+		err := json.Unmarshal([]byte(o.Value), &m)
+		if err != nil {
+			return identity, fmt.Errorf("unable to unmarshal json for value")
+		}
+		identity.Unknown = m
+	}
+	return identity, nil
+}
+
+// Identity - union type, used to access the correct originating identity
+// implementation type
+type Identity struct {
+	Platform     string
+	Kubernetes   *UserInfo
+	CloudFoundry map[string]interface{}
```

Comment:
I would say we create a struct similar to the `UserInfo` with the field `user_id` and `extra` or `leftover` field, that will hold all the leftovers when unmarshmaling the from the json. WDYT? This way we adhere to the spec but also have clear separation between data types.